backend/substrapp/tasks/tasks.py

```diff
@@ -227,25 +227,21 @@ def fetch_model(channel_name, parent_tuple_type, authorized_types, input_model,
 def fetch_models(channel_name, tuple_type, authorized_types, input_models, directory):
 
     models = []
+    exceptions = []
 
     for input_model in input_models:
-        proc = ExceptionThread(target=fetch_model,
-                               args=(channel_name, tuple_type, authorized_types, input_model, directory))
-        models.append(proc)
+        args = (channel_name, tuple_type, authorized_types, input_model, directory)
+        proc = Process(target=fetch_model, args=args)
+        models.append((proc, args))
         proc.start()
 
-    for proc in models:
+    for proc, args in models:
         proc.join()
+        if proc._popen.returncode != 0:
+            exceptions.append(Exception(f'fetch model failed for args {args}'))
```

Comment:
Why do you create a list of exceptions ?

You could create a list of errors directly as an Exception is already raised line 244.

It would be great to add also the return code value in the error message.

Reply:
I keep the old behavior: raise an Exception with the list of exceptions in each parallel job.
Return code is useless as Process log already the python Traceback :)

Reply:
I don't think it makes sense to encapsulate the error messages in exceptions, but that's fine for me.